src/manual/using/actions/schema.yml

```diff
@@ -25,71 +25,117 @@ properties:
     description: |
       List of actions that can be triggered.
     items:
-      type: object
-      properties:
-        name:
-          type: string
-          maxLength: 255
-          description: |
-            The name of this action.  This is used by user interfaces to
-            identify the action. 
-        title:
-          type: string
-          maxLength: 255
-          description: |
-            Title text to be displayed on the button or link triggering the action.
-        description:
-          type: string
-          maxLength: 4096
-          description: |
-            Human readable description of the action in markdown. This
-            can be displayed in a tooltip, popup and/or dialog when triggering
-            the action.
-        kind:
-          enum:
-            - task
-          description: |
-            Specifies the kind of action this is.
-        context:
-          type: array
-          default: []
-          items:
-            type: object
-            additionalProperties:
+      allOf:
```

Comment:
I propose refactoring the schema into:
```yaml
oneOf:
  - type: object
    properties:
      kind: {enum: task}
    ...
  - type: object
    properties:
      kind: {enum: hook}
    ...
```
even the two kinds share some properties, just duplicate those properties. It's okay to document them twice.
The `allOf` + `anyOf` refactoring is harder to do in your head.

Reply:
Another way to reference definitions with `$ref`, this is okay if we render it using a viewer...

Putting definitions to be referenced under a top-level property `definitions: {"mydef": ....}` is convention in json schemas. Then reference it with `$ref: '#/definitions/mydef'`